TFT/src/User/API/Printing.c

```diff
@@ -375,67 +418,58 @@ void printStart(FIL * file, uint32_t size)
   // we assume infoPrinting is clean, so we need to set only the needed attributes
   infoPrinting.size = size;
   infoPrinting.printing = true;
+  initPrintSummary();  // init print summary
+
+  if (GET_BIT(infoSettings.send_gcodes, SEND_GCODES_START_PRINT))
+  {
+    sendPrintCodes(0);
+  }
 
   switch (infoFile.source)
   {
-    case BOARD_SD_REMOTE:  // present just to make the code robust. It should never been executed
+    case REMOTE_HOST:      // present just to make the code robust. It should never be executed
+    case BOARD_SD_REMOTE:
+      return;
+
     case BOARD_SD:
-      //infoHost.printing = true;  // Not so fast! Let Marlin tell that he's done!
-      request_M27(infoSettings.m27_refresh_time);  // check if there is a print running from onboard SD or remote host (USB)
+      infoHost.printing = true;
```

Comment:
I wouldn't re-enable this. I left it there commented with a purpose. Maybe I commented it wrong, it should have been "Not so fast! Let Marlin tell that he started printing!"

Reply:
In case of "remote onboard SD" or "remote host" (new octoprint support) printing it is now executed (on parseAck) the function printRemoteStart (which also set infoHost.printing).
Function printStart (the one you are referring to) is instead executed printing from TFT or onboard SD (so TFT SD, TFT USB and onboard SD requested by TFT). Previously there was also a bug with "onboard SD" printing. It was (correctly) executed printStart and then printRemoteStart (wrongly) (that also changed infoFile.source to ONBOARD_SD_REMOTE)

Reply:
As far as I know BOARD_SD is when Marlin prints from the SD card of the motherboard. Let Marlin tell to the TFT that he started the print. In case Marlin dosn't start the print the TFT will wrongly assume that the print has been started.

Reply:
but the Print menu was already displayed by the caller of printStart and then again by printRemoteStart that also changed the source to ONBOARD_SD_REMOTE

Reply:
So? That doesn't mean that it is correct only to ASSUME that Marlin started the print. Marlin has started the print only when it says so.
Let's look at it the other way, what happens if you let Marlin to tell the TFT that is started the print and don't just simply assume it did?  Will it do more harm than just blindly assuming Marlin has started the print?

Reply:
if for any reason  (e.g. SD detached from onboard slot) the print cannot be started after M24 was sent, Marlin will provide an error message and, in both scenarios (old and new), you need to abort the print to close the Printing menu.

Reply:
My question remains. What is wrong leaving Marlin to tell the TFT that the print has been started?
If `infoHost.printing` is false that means that Marlin hasn't start the print . Telling the TFT that Marlin has started the print, even if you know that it hasn't, it's just wrong. You can explain whatever you want, it is wrong.